src/transformers/modeling_utils.py

```diff
@@ -436,6 +436,17 @@ def load_state_dict(checkpoint_file: Union[str, os.PathLike]):
             )
 
 
+def set_initialized_submodules(model, state_dict_keys):
+    """
+    Sets the `_is_hf_initialized` flag in all submodules of a given model when all its weights are in the loaded state
+    dict.
+    """
+    for module_name, module in model.named_modules():
+        loaded_keys = [k.replace(f"{module_name}.", "") for k in state_dict_keys if k.startswith(f"{module_name}.")]
+        if len(set(module.state_dict().keys()) - set(loaded_keys)) == 0:
+            module._is_hf_initialized = True
```

Comment:
I was thinking of a recursive solution which would solve it in a single traversal, but I suppose as long as the model doesn't have thousands of keys it should be as fast.

Reply:
Hmm, I keep forgetting that for some reason it's only `named_parameters` provides a recursive method

Reply:
your is perfect.